pkg/metaserver/agent/metric/helper/system.go

```diff
@@ -23,78 +23,76 @@ import (
 		if err != nil {
 			return 0, 0, 0, fmt.Errorf(errMsgGetSystemMetrics, consts.MetricMemFreeSystem, err)
 		}
-		free = data.Value
-		_ = emitter.StoreFloat64(metricsNameSystemMetric, free, metrics.MetricTypeNameRaw,
-			metrics.ConvertMapToTags(map[string]string{
-				metricsTagKeyMetricName: consts.MetricMemFreeSystem,
-			})...)
-
-		data, err = metricsFetcher.GetNodeMetric(consts.MetricMemTotalSystem)
+		total, err = GetNodeMetric(metricsFetcher, emitter, consts.MetricMemTotalSystem)
 		if err != nil {
 			return 0, 0, 0, fmt.Errorf(errMsgGetSystemMetrics, consts.MetricMemTotalSystem, err)
 		}
-		total = data.Value
-		_ = emitter.StoreFloat64(metricsNameSystemMetric, total, metrics.MetricTypeNameRaw,
-			metrics.ConvertMapToTags(map[string]string{
-				metricsTagKeyMetricName: consts.MetricMemTotalSystem,
-			})...)
 	}
 
-	data, err := metricsFetcher.GetNodeMetric(consts.MetricMemScaleFactorSystem)
+	scaleFactor, err = GetNodeMetric(metricsFetcher, emitter, consts.MetricMemScaleFactorSystem)
 	if err != nil {
 		return 0, 0, 0, fmt.Errorf(errMsgGetSystemMetrics, consts.MetricMemScaleFactorSystem, err)
 	}
-	scaleFactor = data.Value
-	_ = emitter.StoreFloat64(metricsNameSystemMetric, scaleFactor, metrics.MetricTypeNameRaw,
-		metrics.ConvertMapToTags(map[string]string{
-			metricsTagKeyMetricName: consts.MetricMemScaleFactorSystem,
-		})...)
 
 	return free, total, scaleFactor, nil
 }
 
 func GetSystemKswapdStealMetrics(metricsFetcher metric.MetricsFetcher, emitter metrics.MetricEmitter) (metricutil.MetricData, error) {
-	kswapdSteal, err := metricsFetcher.GetNodeMetric(consts.MetricMemKswapdstealSystem)
+	return GetNodeMetricWithTime(metricsFetcher, emitter, consts.MetricMemKswapdstealSystem)
+}
+
+func GetNodeMetricWithTime(metricsFetcher metric.MetricsFetcher, emitter metrics.MetricEmitter, metricName string) (metricutil.MetricData, error) {
+	metricData, err := metricsFetcher.GetNodeMetric(metricName)
 	if err != nil {
-		return metricutil.MetricData{}, fmt.Errorf(errMsgGetSystemMetrics, "mem.kswapdsteal.system", err)
+		return metricutil.MetricData{}, fmt.Errorf(errMsgGetSystemMetrics, metricName, err)
 	}
-	_ = emitter.StoreFloat64(metricsNameSystemMetric, kswapdSteal.Value, metrics.MetricTypeNameRaw,
+	_ = emitter.StoreFloat64(metricsNameSystemMetric, metricData.Value, metrics.MetricTypeNameRaw,
 		metrics.ConvertMapToTags(map[string]string{
-			metricsTagKeyMetricName: consts.MetricMemKswapdstealSystem,
+			metricsTagKeyMetricName: metricName,
 		})...)
+	return metricData, nil
+}
 
-	return kswapdSteal, nil
+func GetNodeMetric(metricsFetcher metric.MetricsFetcher, emitter metrics.MetricEmitter, metricName string) (float64, error) {
+	metricWithTime, err := GetNodeMetricWithTime(metricsFetcher, emitter, metricName)
+	if err != nil {
+		return 0, err
+	}
+	return metricWithTime.Value, err
+}
+
+func GetNumaMetric(metricsFetcher metric.MetricsFetcher, emitter metrics.MetricEmitter, metricName string, numaID int) (float64, error) {
```

Comment:
some functions should be exported outside while others not, we should apparently separate them, either by a different file or lowercased functions

Reply:
which one should not be exported outside?